tests/install_noded.sh

```diff
@@ -79,18 +79,24 @@ function maybe_update {
 function calc_pytestinit_nodeimpl_version {
 
     # returns the version of $node_impl from pytest.ini from a line which looks like:
-    # addopts = --bitcoind-version v0.21.1 --elementsd-version v0.20.99
+    # addopts = --bitcoind-version v22.0 --elementsd-version v0.20.99
     local node_impl=$1
-    if cat ../pytest.ini | grep -q "addopts = --${node_impl}d-version" ; then
-        # in this case, we use the expected version from the test also as the tag to be checked out
-        # i admit that this is REALLY ugly. Happy for any recommendations to do that more easy
-        PINNED=$(cat ../pytest.ini | grep "addopts = " | cut -d'=' -f2 |  sed 's/--/+/g' | tr '+' '\n' | grep ${node_impl} |  cut -d' ' -f2)
-       
-        if [ "$node_impl" = "elements" ]; then
-            # in the case of elements, the tags have a "elements-" prefix
-            PINNED=$(echo "$PINNED" | sed 's/v//' | sed 's/^/elements-/')
-        fi
+    if [ "$node_impl" = "bitcoin" ]; then
+        PINNED=v22.0
+    elif [ "$node_impl" = "elements" ]; then
+        PINNED=v0.21.0
```

Comment:
This doesn't look right. It's hardcoding the version here?! I'm aware that the above code is not particulary nice and maybe even not robust. Didn't it work anymore?

Reply:
@k9ert the problem is that v22+ changes the numbering convention. The existing code was parsing v22 into "v22.0.0" (note the extra ".0" at the end) which then fails the binary download because no such version number exists. I didn't have the bash/regex skills to handle both version number formats (v0.21.0 vs v22.0).

Reply:
Really stupid. The `getnetworkinfo()["subversion"]` call is returning `/Satoshi:22.0.0/` however the binaries are skipping one [zero, like 22.0 ](https://bitcoincore.org/bin/bitcoin-core-22.0/).

@Sjors maybe you have an idea why that is? Is it worth filing an issue about? It creates needless friction e.g. here.

Reply:
I opened an issue: https://github.com/bitcoin/bitcoin/issues/23615